scalameta/trees/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -300,6 +300,9 @@ object TreeSyntax {
         case t: Type.Select       => m(SimpleTyp, s(t.qual, kw("."), t.name))
         case t: Type.Project      => m(SimpleTyp, s(p(SimpleTyp, t.qual), kw("#"), t.name))
         case t: Type.Singleton    => m(SimpleTyp, s(p(SimpleExpr1, t.ref), ".", kw("type")))
+        case t: Type.Literal      =>
+          if (dialect.allowLiteralTypes) m(SimpleTyp, s(t.literal))
+          else throw new UnsupportedOperationException(s"$dialect doesn't support literal types")
```

Comment:
If we decide to remove `Type.Literal`, it would be still useful to keep this check.